airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py

```diff
@@ -71,6 +71,10 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
 
             for record in data:
                 yield record
+    
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        if response.status_code == requests.codes.too_many_requests:
+            return float(response.headers.get("X-RateLimit-Reset", 0))
```

Comment:
The default `X-RateLimit-Reset` goes with the value of `67150` or more, depending on when the limit is reached. The general Rate Limits are applied to per User per Day (Daily basis).

Q: Do we need to wait for `18h 40min +`, when this limit is reached? 

Suggestion:  
1) we should also check the `code` property inside of the `response.json()` to find the code == 8, which exactly corresponds to the `Max Rate Limit is reached`
2) skip the sync until the rate limits are restored, therefore we should override the `read_records` to return `[ ]` if such conditions are met.

Reply:
@bazarnov If I am understanding it correctly, instead of overriding the `backoff_time` method (which potentially can wait for too long), should I instead for `response.json()["code"] == 8 and  response.headers()["x-envoy-ratelimited"]` when reading data?

When HTTP status code is 429 and that condition is met, I should return `[]`. But what if we get HTTP 429 and `response.json()["code"] == 8 and  response.headers()["x-envoy-ratelimited"]` does not evaluate to `True`?

I agree on not waiting until rate limit is reset, but want to make sure it does the right thing with whatever response it gets from Pinterest API.

Reply:
The logic of backoff should include 2 scenarios: 
1) when we hit the regular `429 - Too many requests`
2) when we hit `420 - Max Rate limit Reached`

For the 1st - we should wait the min time mentioned in `X-RateLimit-Reset`, for the 2nd - the logic is already present in the latest version, so I believe you need to merge with master and then check the updates for the logic and cover the case 1 and we should be good to go.